mailbox/api/src/main/java/org/apache/james/mailbox/MessageManager.java

```diff
@@ -321,6 +323,8 @@ public Flags getFlags() {
      */
     MessageResultIterator getMessages(MessageRange set, FetchGroup fetchGroup, MailboxSession mailboxSession) throws MailboxException;
 
+    Publisher<ComposedMessageIdWithMetaData> getFlags(MessageRange set, MailboxSession session);
```

Comment:
name doesn't match return type. Please name the method listMetadata or something similar